org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseInitHandler.java

```diff
@@ -113,6 +113,11 @@ public InitializeResult initialize(InitializeParams param) {
 			Preferences prefs = Preferences.createFrom((Map<String, Object>) settings);
 			prefs.setRootPaths(rootPaths);
 			preferenceManager.update(prefs);
+			try {
+				ProjectsManager.setAutoBuilding(preferenceManager.getPreferences().isAutobuildEnabled());
+			} catch (CoreException e) {
+				JavaLanguageServerPlugin.logException("Failed apply autobuilding setting", e);
+			}
```

Comment:
We disable auto building on https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/3a8937a2c90e76b5720c58c2f6edf326164cce64/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/LanguageServerApplication.java#L79
Instead of adding this code, you can remove `LanguageServerApplication.prepareWorkspace()`.
AFAICS, it doesn't affect performance.

Reply:
If I only remove `LanguageServerApplication.prepareWorkspace()`, then it is impossible to disable autobuild through initialization params. I'll still remove it because autobuild should be enabled by default unless configured otherwise.

Reply:
>  I'll still remove it because autobuild should be enabled by default unless configured otherwise.

Eclipse ResourcesPlugin remembers the workspace auto-build settings. See 
- https://github.com/eclipse-platform/eclipse.platform/blob/8fe6bba1fe21e5cd255e5398596b5bc5e34f8342/resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/ResourcesPlugin.java#L214
- https://github.com/eclipse-platform/eclipse.platform/blob/8fe6bba1fe21e5cd255e5398596b5bc5e34f8342/resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/PreferenceInitializer.java#L36 (default)